libraries/AP_RPM/RPM_Pin.cpp

```diff
@@ -50,22 +50,22 @@ void AP_RPM_Pin::update(void)
 {
     if (last_pin != get_pin()) {
         // detach from last pin
-        if (last_pin != (uint8_t)-1 &&
+        if ((last_pin > 0) &&
             !hal.gpio->detach_interrupt(last_pin)) {
-            gcs().send_text(MAV_SEVERITY_WARNING, "RPM: Failed to detach from pin %u", last_pin);
+            gcs().send_text(MAV_SEVERITY_WARNING, "RPM: Failed to detach from pin %d", last_pin);
```

Comment:
remove this and add a have_attached_interrupt bool, and don't detach if we haven't successfully attached

Reply:
txs for the review. done.